run_clang_format.sh

```diff
@@ -0,0 +1,6 @@
+#! /bin/bash
+#
+# Finds all .cc and .h files in code/code/** and runs clang-format on them, editing them in-place
+# using the rules in the .clang-format file in the root folder
+
+find code/code/ -iname '*.h' -o -iname '*.cc' | xargs clang-format -i
```

Comment:
Not really crucial, but fun for the future:
filenames can contain any old junk, including terminal bells, newlines, nonprintables, invalid UTF-8-like sequences, etc (only the null terminator and `/` are not allowed). A newline in filename would break this command.
You can work around this issue by using null characters as the separators between files:
`find ... -print0 | xargs -0 ...`

Reply:
I might as well add this to the script, though hopefully people will use the `pre-commit` tool instead, which accepts specific filetypes as a configuration variable and shouldn't be susceptible to the issue you pointed out with my script. That tool was what I used to format all the files and just included the script as a "might as well".